etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2526 @@
+* @name await-thenable
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [await-thenable]{@link https://typescript-eslint.io/rules/await-thenable}
+*
+* @example
+* // Bad...
+* function test() {
+*   await 123;
+* }
+*
+* // Good...
+* async function test() {
+*   await Promise.resolve();
+* }
+*/
+rules[ '@typescript-eslint/await-thenable' ] = 'error';
+
+/**
+* Disables the prohibition against TypeScript `@ts-` comments.
+*
+* @name ban-ts-comment
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [ban-ts-comment]{@link https://typescript-eslint.io/rules/ban-ts-comment}
+*
+* @example
+* // Okay...
+* // @ts-ignore
+* console.log(someVarThatDoesNotExist);
+*/
+rules[ '@typescript-eslint/ban-ts-comment' ] = 'off';
+
+/**
+* Disables the ban on certain TypeScript types like `{}` or `object`.
+*
+* @name ban-types
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [ban-types]{@link https://typescript-eslint.io/rules/ban-types}
+*
+* @example
+* // Okay...
+* type foo = {};
+* type bar = object;
+*/
+rules[ '@typescript-eslint/ban-types' ] = 'off';
+
+/**
+* Enforces the use of consistent type assertions.
+*
+* @name consistent-type-assertions
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [consistent-type-assertions]{@link https://typescript-eslint.io/rules/consistent-type-assertions}
+*
+* @example
+* // Bad...
+* let x = <number>someValue;
+*
+* // Good...
+* let x = someValue as number;
+*/
+rules[ '@typescript-eslint/consistent-type-assertions' ] = 'error';
+
+/**
+* Enforces consistent usage of type definitions either with `interface` or `type`.
+*
+* @name consistent-type-definitions
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [consistent-type-definitions]{@link https://typescript-eslint.io/rules/consistent-type-definitions}
+*
+* @example
+* // Bad...
+* type T = {};
+*
+* // Good...
+* interface T {}
+*/
+rules[ '@typescript-eslint/consistent-type-definitions' ] = 'error';
+
+/**
+* Disables the rule that requires dot notation when accessing properties.
+*
+* @name dot-notation
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [dot-notation]{@link https://typescript-eslint.io/rules/dot-notation}
+*
+* @example
+* // Okay...
+* let x = foo[ 'bar' ];
+* let y = foo.bar;
+*/
+rules[ '@typescript-eslint/dot-notation' ] = 'off';
+
+/**
+* Requires that functions have an explicit return type.
+*
+* @name explicit-function-return-type
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [explicit-function-return-type]{@link https://typescript-eslint.io/rules/explicit-function-return-type}
+*
+* @example
+* // Bad...
+* function test() {
+*   return 42;
+* }
+*
+* // Good...
+* function test(): number {
+*   return 42;
+* }
+*/
+rules[ '@typescript-eslint/explicit-function-return-type' ] = 'error';
+
+/**
+* Enforces explicit visibility declarations for class members.
+* 'no-public' option requires explicit visibility except for public members.
```

Comment:
Let's move this comment to a `## Notes` section.

Reply:
Done.